__init__.py

```diff
@@ -1078,7 +1078,15 @@ def copyResourceToReportFolder(filename):
                             if self.isWorkstationFirstPass:
                                 filename = filename.replace(".htm", "_ix2.htm" if hasPrivateData else "_ix1.htm")
                         elif sourceZipStream is not None:
-                            with FileSource.openFileSource(_filepath, cntlr, sourceZipStream).file(_filepath, binary=True)[0] as fout:
+                            if filesource.isArchive and _filepath in filesource.dir:
+                                _srcfilepath = os.path.join(filesource.baseurl, _filepath)
+                            else:
+                                _srcfilepath = os.path.join(_xbrldir, _filepath)
+                            if sourceZipStream is not None:
```

Comment:
This code is already within a conditional checking that `sourceZipStream` is not None above. Is this code block supposed to be outside of that?